csrc/keyutils.cpp

```diff
@@ -27,10 +27,13 @@ EVP_PKEY* der2EvpPrivateKey(const unsigned char* der, const int derLen, const bo
     throw_java_ex(javaExceptionClass, "Extra key information");
   }
   if (!pkcs8Key) {
-    throw_openssl(javaExceptionClass, "Unable to parse key");
+    throw_openssl(javaExceptionClass, "Unable to parse DER key into PKCS8_PRIV_KEY_INFO");
   }
   EVP_PKEY* result = EVP_PKCS82PKEY(pkcs8Key);
   PKCS8_PRIV_KEY_INFO_free(pkcs8Key);
+  if (!result) {
+    throw_java_ex(javaExceptionClass, "Unable to convert PKCS8_PRIV_KEY_INFO to EVP_PKEY");
```

Comment:
Please replace `throw_java_ex` with `throw_openssl`. That will properly check the openssl error stack and propagate up the contents of it (if it is not empty).